extensions/io/org.eclipse.smarthome.io.mqttembeddedbroker/src/main/java/org/eclipse/smarthome/io/mqttembeddedbroker/internal/EmbeddedBrokerService.java

```diff
@@ -161,130 +33,16 @@ public void brokerRemoved(String brokerID, MqttBrokerConnection broker) {
-            properties.put(BrokerConstants.PORT_PROPERTY_NAME, BrokerConstants.DISABLED_PORT_BIND);
-            properties.put(BrokerConstants.KEY_MANAGER_PASSWORD_PROPERTY_NAME, "esheshesh");
-        } else {
-            if (port == null) {
-                port = 1883;
-            }
-            // with SSL_PORT_PROPERTY_NAME set, netty tries to evaluate the SSL context and shuts down immediately.
-            // properties.put(BrokerConstants.SSL_PORT_PROPERTY_NAME, BrokerConstants.DISABLED_PORT_BIND);
-            properties.put(BrokerConstants.PORT_PROPERTY_NAME, Integer.toString(port));
-        }
-
-        // Authentication
-        io.moquette.spi.security.IAuthenticator authentificator = null;
-        if (StringUtils.isNotBlank(username) && StringUtils.isNotBlank(password)) {
-            properties.put(BrokerConstants.ALLOW_ANONYMOUS_PROPERTY_NAME, false);
-            properties.put(BrokerConstants.AUTHENTICATOR_CLASS_NAME,
-                    MqttEmbeddedBrokerUserAuthenticator.class.getName());
-            authentificator = new MqttEmbeddedBrokerUserAuthenticator(username, password.getBytes());
-        } else {
-            properties.put(BrokerConstants.ALLOW_ANONYMOUS_PROPERTY_NAME, true);
-        }
-
-        // Persistence:
-
-        // The user may have entered a full path. Check for '/' for Unix based systems and for a
-        // a colon as second character for windows based paths ("C:/abc")
-        if (!persistenceFilename.startsWith("/")
-                && !(persistenceFilename.length() > 2 && persistenceFilename.charAt(1) == ':')) {
-            persistenceFilename = ConfigConstants.getUserDataFolder() + File.separator + persistenceFilename;
-        }
-
-        // The user may have set an empty string on purpose: Disable persistence file
-        if (!persistenceFilename.isEmpty()) {
-            properties.put(BrokerConstants.PERSISTENT_STORE_PROPERTY_NAME, persistenceFilename);
-        }
-
-        // We may provide ACL functionality at some point as well
-        IAuthorizator authorizer = null;
-
-        // Secure connection support
-        // TODO wait for NetworkServerTls implementation
-        // try {
-        // final SSLContext sslContext = networkServerTls.createSSLContext("mqtt");
-        // server.startServer(new MemoryConfig(properties), null, () -> sslContext, authentificator, authorizer);
-        // } catch (GeneralSecurityException | IOException e) {
-        // logger.error("No SSL available", e);
-        server.startServer(new MemoryConfig(properties), null, null, authentificator, authorizer);
-        // }
-
-        this.server = server;
-        metrics.setServer(server);
-        ScheduledExecutorService s = new ScheduledThreadPoolExecutor(1);
-        detectStart.startBrokerStartedDetection(port, s);
-    }
+    void startEmbeddedServer(Integer portParam, boolean secure, String username, String password,
+            @NonNull String persistenceFilenameParam) throws IOException;
```

Comment:
We don't use `@NonNull`, we are using `@NonNullByDefault` on classes / interfaces and @Nullable on return values, arguments, ...